src/js/actions.js

```diff
@@ -7,25 +7,28 @@ import { table } from "./app.js";
 
 //// Left click
 
-export function clickOnCell(event){
+export function clickOnCell(e){
 
-    if (!table.timeCounterIsStart){startTimer()}
+    if (table.won === undefined && !table.timeCounterIsStart){startTimer()}
     
-    let cellId = event.target.id;
+    let cellId = e.target.id;
     let cellElement = document.getElementById(cellId);
+    let cell = table.tableCells[cellId];
+
+    if (!cell || cell.haveFlag || table.won !== undefined) return;
```

Comment:
This is newly introduced to realise event delegation. Same further down in `rightClickOnCell()`